### PR TITLE
[M] Cleanup Reporting.Reporter tool

### PR DESCRIPTION
**Pull request by e-kharion**
Reporting.Reporter tool polished

### CODE REVIEW

File: src/tools/Reporting/Reporting/EnvironmentProvider.cs [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
+using System.Collections;
 
 namespace Reporting;
 
 public class EnvironmentProvider : IEnvironment
 {
     public string GetEnvironmentVariable(string variable) => Environment.GetEnvironmentVariable(variable);
-    public System.Collections.IDictionary GetEnvironmentVariables() => Environment.GetEnvironmentVariables();
+
+    public IDictionary GetEnvironmentVariables() => Environment.GetEnvironmentVariables();
```
Comment (DrewScoggins) [inline src/tools/Reporting/Reporting/EnvironmentProvider.cs:14]: Why move this onto one line, but not the property above?

File: src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public static IConfig Create(
                 .AddValidator(new UniqueArgumentsValidator()) // don't allow for duplicated arguments #404
                 .WithSummaryStyle(SummaryStyle.Default.WithMaxParameterColumnWidth(36)); // the default is 20 and trims too aggressively some benchmark results
 
-            if (Reporter.CreateReporter().InLab)
+            if (new EnvironmentProvider().IsLabEnvironment())
```
Comment (DrewScoggins) [inline src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs:61]: Could we just make this a static method we call on EnvironmentProvider? Save the allocation.

**Review by DrewScoggins — DISMISSED** [review]
I left a few comments, but overall this LGTM.

File: src/tools/Reporting/Reporting/EnvironmentProvider.cs (hunk 1 above)
Comment (e-kharion) [reply src/tools/Reporting/Reporting/EnvironmentProvider.cs:14]: I wanted to insert an empty line between methods.

File: src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs (hunk 2 above)
Comment (e-kharion) [reply src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs:61]: I wanted to have the possibility to mock environment for unit testing - thus the interface and it's implementation. 
At the same time I wanted to refactor only a limited portion code - thus the least possible changes outside the Reporting solution (like here in BenchmarkDotNet.Extensions). However good point with the allocation. I introduced a new static variable of the IEnvironment so the EnvironmentProvider is created only once.

**Review by LoopedBard3 — APPROVED** [review]

**Review by DrewScoggins — APPROVED** [review]

**Pull request merged**